woocommerce/woocommerce
# Translate the labels for units of measure

### All Submissions:


### Changes proposed in this Pull Request:

Introduces data and methods for localizing the labels on units of measure, e.g. `kg` or `cm`, and ensures they are utilized wherever those labels are output on the front end and in the admin UI.

Closes #31486


### How to test the changes in this Pull Request:

Testing this involves looking through the UI (front end and admin) to see if there is anywhere that still shows untranslated unit measure labels. In order to differentiate from labels that _have_ been translated, you can add this snippet as an mu-plugin on your test site:

```php
<?php

add_filter(
	'gettext',
	function( $translated, $original ) {
		$weight_units    = array( 'kg', 'g', 'lbs', 'oz' );
		$dimension_units = array( 'm', 'cm', 'mm', 'in', 'yd' );

		$new_string = $original;

		if ( in_array( $new_string, $weight_units, true ) ) {
			$new_string = "💰";
		} else if ( in_array( $new_string, $dimension_units, true ) ) {
			$new_string = "🦄";
		}

		return $new_string;
	},
	99,
	2
);
```

This will make it so any abbreviated weight unit label that's successfully being translated will show as 💰 and any translated dimensions label will show as 🦄.

Some places to check:
* Single product views
* Product management screens
* Product import/export
* Shipping management screens
* WooCommerce settings

### Other information:


### FOR PR REVIEWER ONLY:
